src/com/company/GameManager.java

```diff
@@ -36,58 +46,44 @@ public static void play() {
         for (int i = 0; i < NUMBER_OF_PLAYERS; i++) {
             ioManager.showMessage(format(FORMATTED_PROVIDE_NAME, PLAYER, i + 1, PROVIDE_YOUR_NAME));
             String name = ioManager.getName();
-
             players[i] = new Player(name, sign[i]);
-
             ioManager.showMessage(format(FORMATED_WELCOME, HELLO, players[i].getName(), PLAYING_AS, players[i].getSign()));
-
         }
+    }
 
+    public void playGame(Player players[]) {
+        boolean shouldPlayAgain = true;
         final int boardSize = ioManager.getBoardSize(MIN_BOARD_SIZE, MAX_BOARD_SIZE);
         ioManager.showMessage(format(FORMATED_SELECT, SELECTED, boardSize));
         final Board board = new Board(boardSize);
-
-        Player currentPlayer = players[0];
-        if (getRandomPlayerIndex(NUMBER_OF_PLAYERS) == 1) {
-            currentPlayer = players[1];
-        }
+        int activePlayerId = getRandomPlayerIndex(NUMBER_OF_PLAYERS);
 
         ioManager.showMessage(DRAWN_PLAYER);
-        ioManager.showMessage(format(FORMATED_PLAYER, currentPlayer.getName(), currentPlayer.getSign()));
+        ioManager.showMessage(format(FORMATED_PLAYER, players[activePlayerId].getName(), players[activePlayerId].getSign()));
 
         do {
             ioManager.showBoard(board);
-            ioManager.showMessage(format(FORMATED_PLAYER, currentPlayer.getName(), currentPlayer.getSign()));
+            ioManager.showMessage(format(FORMATED_PLAYER, players[activePlayerId].getName(), players[activePlayerId].getSign()));
```

Comment:
Czy to prawidłowo, że linijki 62 i 66 są takie same i są prawie zaraz po sobie? 
Ta przed pętlą wydaje się niepotrzebna.

Reply:
to jest informacja który gracz został wylosowany jako pierwszy:
Jako pierwszy gra
ela (O)

Reply:
ok, ale czemu jest dwa razy?
Popatrz na linijki 64 i 69 (teraz się przesunęły dalej) i powiedz mi czym się różnią?
Jak dla mnie wygląda na dwa razy to samo przy pierwszym przebiegu.

Reply:
DRAWN_PLAYER = "Jako pierwszy gra"; czyli pierwsza linijka, przed pętlą podaje wynik losowania gracza. A później rysuje się pole gry i przed oczekiwaniem na wpisanie znaku informuję czyj teraz jest ruch. Przy pierwszym ruchu gdyby pominąć pierwszą linijkę to zabraknie mi informacji że pierwszy gracz został wylosowany, pozostanie wywołanie do tablicy bez wytłumaczenia dlaczego.

Reply:
Ok, spoko, masz rację. Nie rozumiem tylko w takim razie dlaczego 69 i 70 nie są jedną linijką?
Mogłoby być tak:
`ioManager.showMessage(DRAWN_PLAYER + format(FORMATED_PLAYER, players[activePlayerId].getName(), players[activePlayerId].getSign()));`